### PR TITLE
Basic Cypress Testing

### PR DESCRIPTION
**Pull request by nico-bellante**

### CODE REVIEW

File: packages/editor/cypress/integration/scriptlab_spec.js [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+
+    openEditorWithState({
+      sessionStorage: { host },
+      localStorage,
+    });
+
+    openSolutionFromEditor(ReportGeneration.id);
+
+    const textToSet = '// This is a test\n ';
+    clearEditor()
```
Comment (Zlatkovsky) [inline packages/editor/cypress/integration/scriptlab_spec.js]: I think you should stick some "awaits" in here...

File: packages/editor/cypress/integration/scriptlab_spec.js [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+    );
+
+    openEditorWithState({
+      sessionStorage: { host },
+      localStorage,
+    });
+
+    openSolutionFromEditor(ReportGeneration.id);
+
+    const textToSet = '// This is a test\n ';
```
Comment (Zlatkovsky) [inline packages/editor/cypress/integration/scriptlab_spec.js]: why the extra space?

File: packages/editor/cypress/integration/scriptlab_spec.js [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+    clearEditor()
+      .then(() => editTextInMonacoEditor(textToSet))
+      .then(() => formatDocument());
+
+    openSolutionFromEditor(BlankExcel.id);
+    cy.wait(1000);
+    openSolutionFromEditor(ReportGeneration.id);
+    cy.wait(1000);
+    getEditorValue().then(value => expect(value).to.equal(textToSet.trim()));
```
Comment (Zlatkovsky) [inline packages/editor/cypress/integration/scriptlab_spec.js]: You don't need to trim when you have the right text in the first place...

File: packages/editor/cypress/integration/scriptlab_spec.js [hunk 4]
```diff
@@ -0,0 +1,280 @@
...
+    );
+
+    openEditorWithState({
+      sessionStorage: { host },
+      localStorage,
+    });
+
+    openSolutionFromEditor(ReportGeneration.id);
+
+    const textToSet = '// This is a test\n ';
```
Comment (Zlatkovsky) [inline packages/editor/cypress/integration/scriptlab_spec.js]: Ditto

File: packages/editor/src/setupTests.ts [hunk 5]
```diff
@@ -1,6 +1,10 @@
+import 'common/lib/polyfills';
 import Enzyme from 'enzyme';
 import Adapter from 'enzyme-adapter-react-16';
 
+// tslint:disable-next-line:no-var-requires
+(window as any).require = require('../public/external/monaco-editor-0-14-3/vs/loader');
```
Comment (Zlatkovsky) [inline packages/editor/src/setupTests.ts]: There's a constant for `monaco-editor-0-14-3`.
`\packages\common\src\package-versions.ts`
`HYPHENATED_PACKAGE_VERSIONS`

File: packages/editor/cypress/integration/scriptlab_spec.js (hunk 1 above)
Comment (nico-bellante) [reply packages/editor/cypress/integration/scriptlab_spec.js]: Cypress chainables, while they look like promises kind of, are not promises, and so awaiting them will not work.

**Pull request merged**